pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -297,6 +303,8 @@ func (p *DefaultProvider) generateNetworkInterfaces(options *amifamily.LaunchTem
 				AssociatePublicIpAddress: options.AssociatePublicIPAddress,
 				DeviceIndex:              aws.Int64(0),
 				Groups:                   lo.Map(options.SecurityGroups, func(s v1.SecurityGroup, _ int) *string { return aws.String(s.ID) }),
+				PrimaryIpv6:              lo.Ternary(p.ClusterIPFamily == corev1.IPv6Protocol, lo.ToPtr(true), nil),
```

Comment:
Can we unit test for this behavior? Adding a few test should help us ensure can keep this feature

Reply:
Also, are we able to validate the behavior in our integration tests? https://github.com/aws/karpenter-provider-aws/blob/main/test/suites/ipv6/suite_test.go

Reply:
Yes, I will look into adding more tests.